docs/1.getting-started/12.upgrade.md

```diff
@@ -5,19 +5,257 @@ navigation.icon: i-ph-arrow-circle-up-duotone
 ---
 
 
-## Upgrading Nuxt 3
+## Upgrading Nuxt
 
 ### Latest release
 
-To upgrade Nuxt 3 to the [latest release](https://github.com/nuxt/nuxt/releases), use the `nuxi upgrade` command.
+To upgrade Nuxt to the [latest release](https://github.com/nuxt/nuxt/releases), use the `nuxi upgrade` command.
 
 ```bash [Terminal]
 npx nuxi upgrade
 ```
 
 ### Nightly Release Channel
 
-To use the latest Nuxt 3 build and test features before their release, read about the [nightly release channel](/docs/guide/going-further/nightly-release-channel) guide.
+To use the latest Nuxt build and test features before their release, read about the [nightly release channel](/docs/guide/going-further/nightly-release-channel) guide.
+
+## Testing Nuxt 4
+
+Nuxt 4 is planned to be released **on or before June 14** (though obviously this is dependent on having enough time after Nitro's major release to be properly tested in the community, so be aware that this is not an exact date).
+
+Until then, it is possible to test many of Nuxt 4's breaking changes on the nightly release channel.
+
+### Opting in to Nuxt 4
+
+First, opt in to the nightly release channel [following these steps](/docs/guide/going-further/nightly-release-channel#opting-in).
```

Comment:
Will this be updated when 3.12 is released?

Reply:
Yes. 👍